lib/splitclient-rb/cache/routers/impression_router.rb

```diff
@@ -0,0 +1,48 @@
+module SplitIoClient
+  class ImpressionRouter
+    attr_reader :router_thread
+
+    def initialize(config)
+      @config = config
+      @listener = config.impression_listener
+      @queue = Queue.new
+      router_thread
+
+      if defined?(PhusionPassenger)
+        PhusionPassenger.on_event(:starting_worker_process) do |forked|
+          router_thread if forked
+        end
+      end
+    end
+
+    def add(impression)
+      @queue.push(impression)
+    end
+
+    def add_bulk(impressions)
+      impressions[:split_names].each do |split_name|
+        @queue.push(
+          split_name: split_name.to_s,
+          matching_key: impressions[:matching_key],
+          bucketing_key: impressions[:bucketing_key],
+          treatment: {
+            label: impressions[:treatments_labels_change_numbers][split_name.to_sym][:label],
+            treatment: impressions[:treatments_labels_change_numbers][split_name.to_sym][:treatment],
+            change_number: impressions[:treatments_labels_change_numbers][split_name.to_sym][:change_number]
+          },
+          attributes: impressions[:attributes]
+        )
+      end
+    end
+
+    private
+
+    def router_thread
```

Comment:
you sure? the link seems outdated.

I see `route_impression` being called on every Store_impressions. Mind helping understand the flow?

```ruby
    def store_impression(split_name, matching_key, bucketing_key, treatment, store_impressions, attributes)
      route_impression(split_name, matching_key, bucketing_key, treatment, attributes) if @config.impression_listener && store_impressions

      return if @config.impressions_queue_size <= 0 || !store_impressions

      @impressions_repository.add(split_name,
        'keyName' => matching_key,
        'bucketingKey' => bucketing_key,
        'treatment' => treatment[:treatment],
        'label' => @config.labels_enabled ? treatment[:label] : nil,
        'time' => (Time.now.to_f * 1000.0).to_i,
        'changeNumber' => treatment[:change_number]
      )
    end
```